app/models/location.rb

```diff
@@ -2,32 +2,43 @@
 #
 # Table name: locations
 #
-#  id              :bigint           not null, primary key
-#  city_town       :string
-#  country         :string
-#  province_state  :string
-#  zipcode         :string
-#  created_at      :datetime         not null
-#  updated_at      :datetime         not null
-#  organization_id :bigint
+#  id             :bigint           not null, primary key
+#  city_town      :string
+#  country        :string
+#  locatable_type :string           not null
+#  province_state :string
+#  zipcode        :string
+#  created_at     :datetime         not null
+#  updated_at     :datetime         not null
+#  locatable_id   :bigint           not null
 #
 # Indexes
 #
-#  index_locations_on_organization_id  (organization_id)
-#
-# Foreign Keys
-#
-#  fk_rails_...  (organization_id => organizations.id)
+#  index_locations_on_locatable  (locatable_type,locatable_id)
 #
 class Location < ApplicationRecord
-  acts_as_tenant(:organization)
+  belongs_to :locatable, polymorphic: true
 
-  validates :country, presence: true, length: {maximum: 50, message: "50 characters maximum"}
-  validates :city_town, presence: true, length: {maximum: 50, message: "50 characters maximum"}
-  validates :province_state, presence: true, length: {maximum: 50, message: "50 characters maximum"}
+  with_options(
+    length: {maximum: 50, message: "50 characters maximum"}
+  ) do |location|
+    location.validates :country, presence: true, if: :province_state_present?
+    location.validates :province_state, presence: true, if: :city_town_present?
+    location.validates :city_town, presence: true
+  end
```

Comment:
I believe there is a way to do this with an attribute we set on the model class as well, e.g., foster_location, and we can pass that attribute in params from the controller, and then conditionally call validations if that attribute is true. 

e.g.,

```
attribute :location_required, :boolean

      with_options if: :location_required do
        validations...
      end
```

then in the controller add `location_required: true` to the build params passed to the model. 

I am not saying we have to do it this way...but it might read cleaner? Your call.